htdocs/core/lib/functions.lib.php

```diff
@@ -12017,7 +12017,7 @@ function forgeSQLFromUniversalSearchCriteria($filter, &$errorstr = '', $noand =
 	$t = preg_replace_callback('/'.$regexstring.'/i', 'dolForgeDummyCriteriaCallback', $filter);
 	$t = str_replace(array('and','or','AND','OR',' '), '', $t);		// Remove the only strings allowed between each () criteria
 	// If the string result contains something else than '()', the syntax was wrong
-	if (preg_match('/[^\(\)]/', $t)) {
+	if (!preg_match('/[^\(\)]/', $t)) {
```

Comment:
The old code looks good.
What value do you have for "$filter" at begin of function ?